Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java

```diff
@@ -877,68 +876,44 @@ public boolean onPreDraw()
 				});
 	}
 
-	private void showMenu(int menuId)
+	private void showMenu(SaveLoadStateFragment.SaveOrLoad saveOrLoad)
 	{
-		Fragment fragment;
-
-		switch (menuId)
-		{
-			case SaveStateFragment.FRAGMENT_ID:
-				fragment = SaveStateFragment.newInstance();
-				mSubmenuFragmentTag = SaveStateFragment.FRAGMENT_TAG;
-				break;
-
-			case LoadStateFragment.FRAGMENT_ID:
-				fragment = LoadStateFragment.newInstance();
-				mSubmenuFragmentTag = LoadStateFragment.FRAGMENT_TAG;
-				break;
-
-			default:
-				return;
-		}
-
+		Fragment fragment = SaveLoadStateFragment.newInstance(saveOrLoad);
 		getFragmentManager().beginTransaction()
 				.setCustomAnimations(R.animator.menu_slide_in, R.animator.menu_slide_out)
-				.replace(R.id.frame_submenu, fragment, mSubmenuFragmentTag)
+				.replace(R.id.frame_submenu, fragment, FRAGMENT_SUBMENU_TAG)
 				.commit();
+		mSubMenuVisible = true;
 	}
 
 	private void removeSubMenu()
 	{
-		if (mSubmenuFragmentTag != null)
-		{
-			final Fragment fragment = getFragmentManager().findFragmentByTag(mSubmenuFragmentTag);
+		final Fragment fragment = getFragmentManager().findFragmentByTag(FRAGMENT_SUBMENU_TAG);
 
-			if (fragment != null)
-			{
-				// When removing a fragment without replacement, its animation must be done
-				// manually beforehand.
-				Animations.fadeViewOutToRight(fragment.getView())
-						.withEndAction(new Runnable()
+		if (fragment != null)
+		{
+			// When removing a fragment without replacement, its animation must be done
+			// manually beforehand.
+			Animations.fadeViewOutToRight(fragment.getView())
```

Comment:
I know this is not added by your PR, but since you are cleaning this part I thought I will mention it :)

Most probably this hack was added because of the wrong implementation of the back button of the fragment, I would try to add the fragment to the backStack then in the removal use 
getFragmentManager().popBackStack()

I bet it will work without this hack and the animation will be reversed without any hack

Reply:
Yeah, I glanced at that... Lemme fix it in another commit, I think you're right